collated/test/A0148096W.md

```diff
@@ -0,0 +1,1478 @@
+                td.learnVim, td.buyBirthdayGift, td.signUpForYoga);
+    }
+    
+    @Test
+    public void hide_deadline_byYear2020() {
+        assertListResult("hide by 12 dec 2020", td.submitPrototype, 
+                td.developerMeeting, td.researchWhales, 
+                td.learnVim, td.signUpForYoga);
+    }
+    
+    @Test
+    public void hide_startTime_fromYear2022() {
+        assertListResult("hide from 1 jan 2022", 
+                td.submitProgressReport, td.developerMeeting,
+                td.learnVim, td.buyBirthdayGift, td.signUpForYoga);
+    }
+    
+    @Test
+    public void hide_endTime_toYear2021() {
+        assertListResult("hide to 30 dec 2021");
+    }
+    
+    @Test
+    public void hide_tags_withMatches() {
+        assertListResult("hide tag urgent", td.developerMeeting,
+                td.researchWhales, td.learnVim, 
+                td.buyBirthdayGift, td.signUpForYoga);
+    }
+    
+    @Test
+    public void hide_tags_noMatches() {
+        assertListResult("hide tag thistagdoesnotexist", 
+                td.submitPrototype, td.submitProgressReport, 
+                td.developerMeeting, td.researchWhales,
+                td.learnVim, td.buyBirthdayGift, td.signUpForYoga);
+    }
+    
+
+    //---------------- Tests combined arguments ----------------------
+    
+    @Test
+    public void hide_combinedArgs() {
+        assertListResult("hide uncomplete tasks, from 1 jan 1998"
+                + " to 1 jan 2024, tag urgent");
+    }
+    
+    @Test
+    public void hide_continuously_narrowsList() {
+        assertListResult("hide tag thistagdoesnotexist", 
+                td.submitPrototype, td.submitProgressReport, 
+                td.developerMeeting, td.researchWhales,
+                td.learnVim, td.buyBirthdayGift, td.signUpForYoga);
+        
+        assertListResult("hide tag urgent", td.developerMeeting,
+                td.researchWhales, td.learnVim, 
+                td.buyBirthdayGift, td.signUpForYoga);
+        
+        assertListResult("hide tasks", td.researchWhales,
+                td.learnVim, td.signUpForYoga);
+        
+        assertListResult("hide uncompleted", td.researchWhales);
+        
+        assertListResult("hide completed");
+    }
+    
+    //---------------- Utility methods ----------------------
+
+    private void assertListResult(String command, TestTask... expectedHits ) {
+        commandBox.runCommand(command);
+        assertListSize(expectedHits.length);
+        assertResultMessage(expectedHits.length + " tasks listed!");
+        assertTrue(taskListPanel.isListMatching(expectedHits));
+    }
+}
+```
+###### \java\guitests\ListCommandTest.java
+``` java
+
+package guitests;
+
+import org.junit.Test;
+
+import teamfour.tasc.testutil.TestTask;
+
+import static org.junit.Assert.assertTrue;
+
+public class ListCommandTest extends AddressBookGuiTest {
+
+    /*
+     * All exceptions for invalid arguments are handled 
```

Comment:
what do you mean by this? if it is no longer relevant, please remove this comment.
similarly above as well